src/App.tsx

```diff
@@ -1,61 +1,120 @@
+/* eslint-disable @typescript-eslint/indent */
+import { useState } from 'react';
+
 import './App.scss';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ToDo } from './types/todo';
+import { findUserById, getTodosMaxId } from './service';
+
+export const App: React.FC = () => {
+  const tasks = todosFromServer.map(todo => {
+    return {
+      ...todo,
+      user: findUserById(todo.userId),
+    };
+  });
+
+  const [todos, setTodos] = useState<ToDo[]>([...tasks]);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setErrorTitle(false);
+    setErrorUser(false);
+  };
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = event => {
+    event.preventDefault();
+
+    setErrorTitle(!title);
+    setErrorUser(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    setTodos([
+      ...todos,
+      {
+        id: getTodosMaxId(todos) + 1,
+        title: title,
+        completed: false,
+        userId: Number(userId),
+        user: findUserById(Number(userId)),
+      },
+    ]);
+
+    resetForm();
+  };
+
+  const handleUserIdChange: React.ChangeEventHandler<
+    HTMLSelectElement
+  > = ev => {
+    setUserId(+ev.target.value);
```

Comment:
The `userId` state is initialized as a number, but the value from the select input is a string. Consider using `parseInt(ev.target.value, 10)` to ensure `userId` remains a number.